machine.go

```diff
@@ -30,6 +31,49 @@ func mergedUsrSystem() bool {
 	return false
 }
 
+// Parse modinfo result and get path of module specified by name
+func getModPath(modname string) string {
+	out, err := exec.Command("modinfo", modname).Output()
+	if err != nil {
+		return ""
+	}
+
+	scanner := bufio.NewScanner(strings.NewReader(string(out)))
+	scanner.Split(bufio.ScanLines)
+	for scanner.Scan() {
+		field := strings.Split(strings.TrimSpace(scanner.Text()), ":")
+		if strings.TrimSpace(field[0]) == "filename" {
+			return strings.TrimSpace(field[1])
+		}
+	}
+	return ""
+}
+
+// Parse modinfo result and get dependencies of module
+// Returns slice of module name
+func getModDepends(modname string) []string {
+	out, err := exec.Command("modinfo", modname).Output()
+	if err != nil {
+		return nil
+	}
+
+	scanner := bufio.NewScanner(strings.NewReader(string(out)))
+	scanner.Split(bufio.ScanLines)
+	var deplist []string
+	for scanner.Scan() {
+		fields := strings.Split(strings.TrimSpace(scanner.Text()), ":")
+		if strings.TrimSpace(fields[0]) == "depends"  {
+			t := strings.TrimSpace(fields[1])
+			if t != "" {
+				list := strings.Split(t, ",")
+				deplist = append(deplist, list...)
+			}
+		}
+	}
+	return deplist
+}
+
```

Comment:
I would make a helper `getModDetails(modname string, dataname string) []string`. And a one-line wrapper for each of filename, depends, etc.

Reply:
sure, will update